include/thread_pool/worker.hpp

```diff
@@ -36,9 +38,9 @@ class Worker
     /**
      * @brief start Create the executing thread and start tasks execution.
      * @param id Worker ID.
-     * @param steal_donor Sibling worker to steal task from it.
+     * @param workers Sibling workers for performing round robin work stealing.
      */
-    void start(size_t id, Worker* steal_donor);
+    void start(size_t id, WorkerVector* workers);
```

Comment:
Why not `std::shared_ptr<WorkerVector>`?

Reply:
Didn't need shared ownership semantics, the root thread pool object owns the worker vector. Also didn't want to worry about the overhead of the reference counting.